newIDE/app/src/MainFrame/index.js

```diff
@@ -935,6 +936,15 @@ export default class MainFrame extends React.Component<Props, State> {
   save = () => {
     if (!this.state.currentProject) return;
 
+    /// Iterate through each opened scene tab and store its settings
+    this.state.editorTabs.editors.forEach(editor => {
+      if (editor.editorRef.constructor.name === 'SceneEditorContainer') {
```

Comment:
This is quite fragile because if I rename SceneEditorContainer, this will break, and the developer can't notice it immediately.

This is also breaking the encapsulation, by digging into the editor. While MainFrame is supposed to handle a bunch of tabs, not even knowing what is inside, we're now making assumption about the whole structure. This is a bit "hacky".
This is also not working for external layouts ;)

The solution, and the contrary to the problem of breaking "encapsulation" is abstraction and giving responsibility to components.
You want to save settings of an editor? Then require the editors to implement a "saveUISettings" method ;)

Then when you save the game (i.e, here):
getEditors(this.state.editorTabs).forEach(editor => editor.editorRef.saveSettings())

Actually, if you don't want to force all editors to implement saveSettings, test for its presence. Check how it's done for closing tabs for example related to an extension: 

https://github.com/4ian/GDevelop/blob/c00e98639a8cae8e911490bc81b9399976b420ff/newIDE/app/src/MainFrame/EditorTabsHandler.js#L190-L193

If possible, add you method in this file, and just call it from MainFrame.js (MainFrame.js is the complete editor so it's becoming very large, better have specific logic like this in other files).

For implementing saveSettings, it's basically how you have done here :) But in SceneEditor (or in other editors) you'll already have the layout, so no need to do a `editor.editorRef.getLayout` ;) 

Finally, call saveSettings at some other moments, like when the tab is closing, and you're done :)

Reply:
I thought that it's a bit hacky too :) Just wanted to get it to do it with least amount of code.

Will refactor this after I sort out the paths bug. Thank you for explaining

Reply:
Ok I think it should be much better now, paving the way to save more of the UI settings in the project file.